packages/cli/src/Server.ts

```diff
@@ -161,11 +161,56 @@ import { getLicense } from '@/License';
 import { corsMiddleware } from './middlewares/cors';
 
 require('body-parser-xml')(bodyParser);
+const typeis = require('type-is');
 
 const exec = promisify(callbackExec);
 
 export const externalHooks: IExternalHooksClass = ExternalHooks();
 
+function parseNdjson(str: string): any[] {
+	const result: any[] = [];
+	const items = str
+		.split('\n')
+		.map((item: string) => item.trim())
+		.filter((item: string) => item !== '');
+	for (const item of items) {
+		try {
+			result.push(JSON.parse(item));
+		} catch (e) {
+			throw Error(
+				// eslint-disable-next-line prettier/prettier
+				`error parsing ndjson, while parsing item ${result.length + 1} of ${result.length}: ${e.message}`,
+				{ cause: e },
+			);
+		}
+	}
+	return result;
+}
+
```

Comment:
```
function parseNdjson(str: string): any[] {
  const items = str.trim().split('\n');
  return items.map((item, i) => {
    try {
      return JSON.parse(item);
    } catch (e) {
      throw new Error(
        `error parsing ndjson, while parsing item ${i + 1} of ${items.length}: ${e.message}`,
      );
    }
  });
}
```


version of the code is equivalent to the original code, but it uses the map method to transform the array of strings into an array of parsed JSON objects, rather than using a for loop. This can make the code easier to read and understand. Additionally, the trim method is used to remove any leading or trailing whitespace from the input string, and the filter method is removed since it is not necessary. Finally, the error message is modified to correctly reference the current item being parsed in the map callback.

Reply:
Thanks!
The code looks better indeed, more readable and there's less memory allocation - trimming each line allocates a new string.

About removing the filter -
an empty line is acceptable according to the spec:
> The parser MAY silently ignore empty lines, e.g. \n\n. This behavior MUST be documented and SHOULD be configurable by the user of the parser.

From my experience, API providers WILL do stupid things like putting empty-lines in the middle of an ndjson, occasionally or all the time (reminder - I opened this PR because of an API provider that did not adhere to standard).
It would be a shame that the n8n user get stuck because of this.

So I think it's better to filter out empty-lines nevertheless.
I wish there was a `str.isEmpty()` that does not allocates a new string.

Reply:
Yes, I agree, it would be fab to have a str.isEmpty() method that does not allocate a new string, but for now, I think, using the trim method is a good solution to avoid unnecessary memory allocation. No?

Reply:
What about what i wrote about filtering out empty lines?
Do you agree?